module/shop_beta/shop_reward.py

```diff
@@ -0,0 +1,25 @@
+from module.base.base import ModuleBase
+from module.logger import logger
+from module.shop.shop_core import CoreShop
+from module.shop.shop_general import GeneralShop
+from module.shop.shop_guild import GuildShop
+from module.shop.shop_medal import MedalShop
+from module.shop.shop_merit import MeritShop
+
+
+class RewardShop(ModuleBase):
+    def run_frequent(self):
+        GeneralShop(self.config, self.device).run()
+
+        self.config.task_delay(server_update=True)
+
+    def run_once(self):
+        MeritShop(self.config, self.device).run()
+
+        CoreShop(self.config, self.device).run()
+
+        GuildShop(self.config, self.device).run()
+
+        MedalShop(self.config, self.device).run()
```

Comment:
Move the page switchings here, should save some time
```python
self.ui_goto_shop()

self.shop_bottom_navbar_ensure(left=X)
MeritShop(self.config, self.device).run()

self.shop_bottom_navbar_ensure(left=X)
CoreShop(self.config, self.device).run()

self.shop_bottom_navbar_ensure(left=X)
GuildShop(self.config, self.device).run()

self.ui_goto_gacha()
self.shop_bottom_navbar_ensure(left=X)
MedalShop(self.config, self.device).run()
```

Reply:
I had placed them in their individual `run()` since I wanted to save time on transitioning to the page if it didn't need to like having an empty filter.  For example, all shops but medal have empty filter or vice versa, no need to go to gacha page if the medal filter is empty. 

But if moving to the shop and then checking the filter and finding out its empty is acceptable, I can update it.  It's probably a fair assumption that the above case would not happen. Even more unlikely enabling the task but with all filter text being empty.  Just like to account for these cases even if the likeliness is low.

Reply:
Move the checks here too, like:
```python
shop = MeritShop(self.config, self.device)
if not shop.filter_empty():
    self.shop_bottom_navbar_ensure(left=X)
    shop.run()
```
This should save time in both cases, having and not having an empty filter

Reply:
Ok I think bringing them out is fine, it would satisfy my concern but I think missing 1 line
```
shop = MeritShop(self.config, self.device)
if not shop.filter_empty():
    self.ui_goto_shop()
    self.shop_bottom_navbar_ensure(left=X)
    shop.run()
```
If outside the condition then would go to shop implicitly so waste transition time going to there if 3/4 are empty (particularly merit, core, and guild since all are in `page_munitions`)

`ui_goto_shop` uses `ui_ensure` so doesn't waste time in taking screenshots if already there.

Reply:
Yeah ,that was missing.